### PR TITLE
feat(web-app): functions catalog update

### LINKED ISSUE #886: Remove thumbnails from functions catalog & update mobile view for all catalogs
Design issue #449 

Redlines for functions catalog 
https://ibm.box.com/s/w6xjolukzbiv9jhqa8di207d7rfy96j4

Note: these changes need made at the functions catalog level, not at the catalog item level if type function, because when there are catalogs with mixed types (e.g. collections), we'll still display placeholder thumbnails for functions so there is visual consistency.
- [x] Remove grid/list switcher so it's only list view
- [x] Remove thumbnail from list view
- [x] When removing thumbnail, set height of the rows
- [x] Still use aspect ratio at `sm` breakpoint

Update for ALL catalogs
- [x] For both tiles with a thumbnail and without a thumbnail, small break points are the same. The status moves to upper right and tags move to lower left.
- [x] For a tile with a thumbnail, at medium breakpoints, the status moves to upper right and tags move to lower left.
- [x] For a tile without a thumbnail, at medium breakpoints, the status and tags stay in place. There is room for the tags to stay in the upper right.

### Here are the specs for a catalog with a thumbnail:

**medium:**
image

**small breakpoint:**
content section same as medium but remove thumbnail
image 

### Catalog tile without a thumbnail:

**medium break point:**
image

**small break point:**
image

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #886

Functions catalog updates, default to list view only with no thumbnails.
Updated design tweaks at mobile/tablet for all catalogs.

#### Changelog

**Changed**

- Add showImage prop to AssetsCatalog
- Default to list view if showImage is false
- Alternate styling if no image present at tablet

#### Testing / reviewing

Test Functions and Component catalog matches spec in #886 at all breakpoints
http://localhost:3000/assets/functions
http://localhost:3000/assets/components

|         | Functions           | Components  |
| ------------- |:-------------:|:-----:|
|  Desktop     | Screen Shot 2022-06-22 at 2 37 02 PM | Screen Shot 2022-06-22 at 2 36 49 PM |
| Tablet      | Screen Shot 2022-06-22 at 2 37 23 PM      | Screen Shot 2022-06-22 at 2 37 36 PM |
| Mobile | Screen Shot 2022-06-22 at 2 38 02 PM     |    Screen Shot 2022-06-22 at 2 37 50 PM |

### CODE REVIEW

File: services/web-app/pages/assets/functions.js [hunk 1]
```diff
@@ -33,8 +33,7 @@ const Functions = ({ librariesData }) => {
     <>
       <NextSeo {...seo} />
       <PageHeader bgColor={func.bgColor} title={seo.title} pictogram={func.icon} />
-      {/* this probably eventually gets replaced  by a functionsCatalog */}
```
Comment (alisonjoseph) [inline services/web-app/pages/assets/functions.js:36]: @francinelucca I didn't see the need for a separate functions catalog component here, but let me know if I missed something!

**Review by kingtraceyj — APPROVED** [review]
looks good to go from the screenshots!

File: services/web-app/pages/assets/functions.js (hunk 1 above)
Comment (francinelucca) [reply services/web-app/pages/assets/functions.js:36]: you're right! not needed 👯

**Review by andreancardona — APPROVED** [review]
🔥

**Pull request merged**